include/ttmlir/Dialect/TTIR/IR/TTIROps.td

```diff
@@ -167,6 +167,13 @@ def TTIR_AbsOp: TTIR_ElementwiseUnaryOp<"abs"> {
     }];
 }
 
+def TTIR_TypecastOp: TTIR_ElementwiseUnaryOp<"typecast"> {
+    let summary = "Eltwise cast op.";
+    let description = [{
+      Eltwise cast operation.
+    }];
+}
```

Comment:
`TTIR_ToLayoutOp` is intended to support typecast.  @rpavlovicTT do you think it's still worth having an explicit op for it? I'm kind of on the fence, ttmetal backend could canonicalize this op to just be a ToLayout, so it's not a big deal.

Reply:
`TTIR_ToLayoutOp` requires `Layout` attribute. Once we introduce `Layout` then we have to propagate it everywhere which seems bit excessive at `TTIR` level graph. That's why I introduced `TTIR_TypecastOp`. I had initially lowered `TTIR_TypecastOp` to `TTNN_ToLayoutOp` (I wasn't aware of `ttnn::typecast`) but it didn't produce correct results in some cases, whereas using `ttnn::typecast` produces correct results for all my test cases.

Reply:
My feeling is that `ToLayoutOp` is getting kinda overloaded with all these responsibilities, isn't it? I'd rather be explicit here if we have a chance with TTNN.

Reply:
I agree with both of your points.  Let's go with this approach.